six/modules/c++/six.sicd/tests/test_add_additional_des.cpp

```diff
@@ -0,0 +1,256 @@
+/* =========================================================================
+* This file is part of six.sicd-c++
+* =========================================================================
+*
+* (C) Copyright 2004 - 2016, MDA Information Systems LLC
+*
+* six-c++ is free software; you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as published by
+* the Free Software Foundation; either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public
+* License along with this program; If not,
+* see <http://www.gnu.org/licenses/>.
+*
+*/
+
+#include <cstdio>
+#include <six/NITFWriteControl.h>
+#include <six/Types.h>
+#include <six/XMLControlFactory.h>
+#include <six/sicd/Utilities.h>
+
+namespace
+{
+void validateArguments(int argc, char** argv)
+{
+    if (argc != 2)
+    {
+        std::string message = "Usage: " + sys::Path::basename(argv[0])
+                + " <XML pathname>";
+        throw except::Exception(Ctxt(message));
+    }
+    if (!str::endsWith(argv[1], ".xml"))
+    {
+        std::string message = "Extension suggests input may not be an XML file. "
+            "Expecting '*.xml'.";
+        throw except::Exception(Ctxt(message));
+    }
+}
+
+std::vector<sys::Int16_T> generateBandData(const six::sicd::ComplexData& data)
```

Comment:
It looks like this gets optimized out for MSVC since 2005 and gcc since 2002 (v. 3.1.1). But should I make it an output parameter instead just in case?

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Thursday, May 26, 2016 9:25 PM
To: ngageoint/six-library <six-library@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>; Author <author@noreply.github.com>
Subject: Re: [ngageoint/six-library] Add method to allow adding extra DES to NITFWriteControl (#76)


In six/modules/c++/six.sicd/tests/test_add_additional_des.cpp<http://cp.mcafee.com/d/2DRPoOd6Qm766kkSmhNKVJBxAsed7bXVJYsOMOyqejhOrhKZtVMQsTd7bNJ55B5dYQsKcfCN6IdCjS7a_Ia0Fl3mWsKrlgRKDbCTrFWeo7fZvzhOYyOVRXBQSkkuhujpsjVqWtAklrEKsG7DR8OJMddECSjt-hsd7bOabatTztPpesRG9px6kOrp3BYwhQzZpgDB91Vs-h2eNboISuvwVM349kITiAGM5oKcfcecFECO5mUm-w0nC7C4miXb2eBN6IdCjS7a_Ia7aEqTjBMlgy9YhBcCSgVsSUOCUr1vF6y0QJM8Wh-IEjOAwYKv8x7oBziWq81110Qga_cDl0QgjGq8dffGod46E8iwhd47xYUWq80Uq63h1eFEw1wq2hs1oODUBm4URfd41EE0MsUqajpEVKrC-b7>:

> +{

> +    if (argc != 2)

> +    {

> +        std::string message = "Usage: " + sys::Path::basename(argv[0])

> +                + " <XML pathname>";

> +        throw except::Exception(Ctxt(message));

> +    }

> +    if (!str::endsWith(argv[1], ".xml"))

> +    {

> +        std::string message = "Extension suggests input may not be an XML file. "

> +            "Expecting '*.xml'.";

> +        throw except::Exception(Ctxt(message));

> +    }

> +}

> +

> +std::vector<sys::Int16_T> generateBandData(const six::sicd::ComplexData& data)

This doesn't matter a ton b/c it's just a test program, but this will potentially result in a copy depending on how smart the compiler is (C++11-compiler will likely optimize this out).

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<http://cp.mcafee.com/d/FZsS91NJ5xNxB5dBAsrKrpop73zhO--rv7cIcECzAQsCQrLnusd7dPhOYrhhphjvd7bz3VIhH3pAZxOLX2walgRKDbCRkdrFOVJSWuzC1P_nUQsL8IKtuVtdB57AnASn4-mKDp55mWbDaxVZicHs3jqpJATvAn3hOYyyODtUTsSjDdqymohBcCSgVv84t8_mk9VigunfAgzIiSbdDDUesNSh_QNdwseoIwNuZA7td7dQ7SbFE_pRx5x10wz-FLLcWg0Obz3P3zaq9IxlK5LE05VxVx5AKOMzFshH3pAZxOLX2xOG6JQVs5k8yv4pj9JAendKcFK6MnWhEwdbs2eAvHa4YF8fbDO8hS9oQKCy0gggd42LP9Rgd44WCy3jPWC3h1G24E4jh1UveeCy0e6xwQgjGq80o6wAn0mcF-9lxedjPh0qa0c7e6yASqerCZTE1VsQJ>, or mute the thread<http://cp.mcafee.com/d/1jWVIg3xAidEIeccEFIIzztPrb38UsqenTPrUVBxB4QsCzASztWXPxEVKqenzqabaarVEVsovdydorcDIel_ok1iG6JQVsSGxHtendKTjQsMevW_6zBV5BPHTbFIEEYyYCOUDORQX8EGThsVkffGhBrwqrpdICXYyUqenAkmkXL6XCOsVHkiP2cFASO7bV2IVeOJIVlxj6Ukr8kJ6h3Y41s5y073zMI6kmgmS8yEs2MACP8QsAge6z1wbCiOP9mzUIbBi8n8W12M4zr0quy1hVIxlK5LE05VxVx5AKOMzFshH3pAZxOLX2xOG6JQVs5k8yv4pj9JAendKcFK6MnWhEwdbs2eAvHa4YF8fbDO8hS9oQKCy0gggd42LP9Rgd44WCy3jPWC3h1G24E4jh1UveeCy0e6xwQgjGq80o6wAn0mcF-9lxedjPh0qa0c7e6yASqerCYhliTwOUrLmV>.


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.